docs/design/proposal/dependency-upgrades/aws-sdk-upgrades.md

```diff
@@ -0,0 +1,152 @@
+in specific AWS service API + documentation). See this
+[v1.42.0 changelog](https://github.com/aws/aws-sdk-go/releases/tag/v1.42.0) as an example
+
+## Scope
+This documentation will focus on automating the updates for aws-sdk-go minor version
+upgrades. Minor version upgrades will be done starting from ACK runtime to code-generator
+and then to all service-controllers using existing automation.
+
+Another [document](./aws-sdk-patch-version-upgrades.md) will focus on how to upgrade
+individual service controllers for the patch releases because Patch releases will only
+impact specific service controllers not the ACK runtime behavior (except IAM service
+changes)
+
+## High Level Overview
+
+The automation will be divided in following sub-parts
+
+### Trigger
+Trigger is responsible for deciding when to upgrade ACK core libraries and service
+controllers with the new aws-sdk-go version. The trigger can be event-driven(tracking
+a new release of aws-sdk-go using webhooks or other methods) or periodic.
+
+### Verification
+Verification is responsible for providing the preview of changes caused by
+aws-sdk-go upgrade. Example: Which controller(s) build successfully with the
+new aws-sdk-go version? Does a new resource/ new fields gets added as part
+of controller build? etc...
+
+### Release
+The actual process of upgrading the go.mod files with latest aws-sdk-go version
+and raising PR.
+
+## Implementation
+
+### Trigger
+For the trigger, the periodic job option should be preferred that will
+run every day. 24 hours frequency is good enough to not delay verifying
+the new aws-sdk-go release. 
+
+Since immediate aws-sdk-go upgrades are not the requirement, the
+Trigger does not need to handle complexity of being event driven.
+
+### Verification
+This will be the major part of this new automation. This step will include
+testing all the service controllers locally with new aws-sdk-go release.
+The testing will include updating the dependency, generating new controller
+source code, compiling and publishing a report of the upgrade. 
+The report will include details like,
+* Which controllers were successfully built and unit tested
+* Which controllers failed to build and compile 
+* The summary of files added/changed for each successful controller. This
+will give an idea where new resources/fields were added in new aws-sdk-go
+release
+
+###### Setup
+There will be a new periodic prowjob which will run once every 24 hours.
+This job will checkout all the service controller repos, along with latest
+ACK runtime release and corresponding code-generator release.
+
+> NOTE: We will not checkout `main` branch of ACK core libraries because this
+job should only validate the changes caused by aws-sdk-go upgrades for
+the latest controller release
+
+###### Controller Build
+* In the periodic job mentioned above, the ENTRYPOINT script will first
+validate whether there is a new minor version upgrade release for aws-sdk-go
```

Comment:
Regarding my comment on filtering above, I think this is a bit too late to check for minor vs. patch. Probably better to just check asap.